contracts/zero-ex/contracts/src/features/OtcOrdersFeature.sol

```diff
@@ -0,0 +1,443 @@
+
+        // Clamp the taker token fill amount to the fillable amount.
+        takerTokenFilledAmount = LibSafeMathV06.min128(
+            takerTokenFillAmount,
+            order.takerAmount
+        );
+        // Compute the maker token amount.
+        // This should never overflow because the values are all clamped to
+        // (2^128-1).
+        makerTokenFilledAmount = uint128(LibMathV06.getPartialAmountFloor(
+            uint256(takerTokenFilledAmount),
+            uint256(order.takerAmount),
+            uint256(order.makerAmount)
+        ));
+
+        if (wethOptions == WethOptions.WrapEth) {
+            require(
+                order.takerToken == WETH,
+                "OtcOrdersFeature/INVALID_WRAP_ETH"
+            );
+            // Wrap ETH
+            WETH.deposit{value: takerTokenFilledAmount}();
+            // Transfer WETH to maker
+            WETH.transfer(order.maker, takerTokenFilledAmount);
+            if (takerTokenFilledAmount < msg.value) {
+                // Refund unused ETH
+                _transferEth(
+                    msg.sender,
+                    msg.value - uint256(takerTokenFilledAmount)
+                );
+            }
+        } else {
+            // Transfer taker -> maker
+            _transferERC20Tokens(
+                order.takerToken,
+                taker,
+                order.maker,
+                takerTokenFilledAmount
+            );
+        }
+
+        if (wethOptions == WethOptions.UnwrapWeth) {
+            require(
+                order.makerToken == WETH,
+                "OtcOrdersFeature/INVALID_UNWRAP_WETH"
+            );
+            // Transfer maker tokens in
+            _transferERC20Tokens(
+                order.makerToken,
+                order.maker,
+                address(this),
+                makerTokenFilledAmount
+            );
+            // Unwrap WETH
+            WETH.withdraw(makerTokenFilledAmount);
+            // Transfer ETH to taker
+            _transferEth(taker, makerTokenFilledAmount);
+        } else {
+            // Transfer maker -> taker.
+            _transferERC20Tokens(
+                order.makerToken,
+                order.maker,
+                taker,
+                makerTokenFilledAmount
+            );
+        }
+    }
+
+    /// @dev Get the order info for an OTC order.
+    /// @param order The OTC order.
+    /// @return orderInfo Info about the order.
+    function getOtcOrderInfo(LibNativeOrder.OtcOrder memory order)
+        public
+        override
+        view
+        returns (LibNativeOrder.OtcOrderInfo memory orderInfo)
+    {
+        // compute order hash.
+        orderInfo.orderHash = getOtcOrderHash(order);
+
+        LibOtcOrdersStorage.Storage storage stor =
+            LibOtcOrdersStorage.getStorage();
+
+        // check tx origin nonce
+        uint256 lastNonce = stor.txOriginNonces
+            [order.txOrigin]
+            [order.txOriginNonceBucket];
```

Comment:
It looks like we support `(2 ** 256) -1` buckets. Is there any value in restricting these buckets to a fixed number? example: 100?

Reply:
no